src/error.rs

```diff
@@ -9,7 +16,59 @@ pub enum HcHttpGatewayError {
     /// Handles configuration parsing errors
     #[error("Configuration error: {0}")]
     ConfigurationError(String),
+    /// Handle path deserialization errors
+    #[error("Path deserialization error: {0}")]
+    PathDeserializationError(#[from] axum::extract::rejection::PathRejection),
+    /// Handle base64 decode errors
+    #[error("Base64 decoding error: {0}")]
+    Base64DecodingError(#[from] base64::DecodeError),
+    /// Handle holo hash errors
+    #[error("HoloHash error: {0}")]
+    HoloHashError(#[from] HoloHashError),
+    /// Handle errors deserializing zome call payload to json
+    #[error("Failed to deserialize JSON to serde_json::Value: {0}")]
+    InvalidJSON(#[from] serde_json::Error),
 }
 
 /// Type aliased Result
 pub type HcHttpGatewayResult<T> = Result<T, HcHttpGatewayError>;
+
+impl IntoResponse for HcHttpGatewayError {
+    fn into_response(self) -> axum::response::Response {
+        match self {
+            HcHttpGatewayError::PathDeserializationError(e) => {
+                tracing::error!("Path deserialization error: {}", e);
+                error_from_status(400, Some("Invalid Request Path"))
+            }
+            HcHttpGatewayError::Base64DecodingError(e) => {
+                tracing::error!("Base64 decode error: {}", e);
+                error_from_status(400, Some("Failed to decode base64 encoded strig"))
+            }
+            HcHttpGatewayError::HoloHashError(e) => {
+                tracing::error!("HoloHash error: {}", e);
+                error_from_status(400, Some("HoloHash error"))
```

Comment:
At the moment this can only be "Invalid base64 DNA hash", I'd rather have that feedback as a user. I think it might make sense to inline some of this error handling to keep the error messages specific where appropriate

Reply:
Updated the error message.

> I think it might make sense to inline some of this error handling to keep the error messages specific where appropriate

Not sure what you mean with this, are you referring to not using the `error_from_status` function building the axum error response for each error variant?

Reply:
> I'm wondering if we should handle the DnaHash error inline rather than converting it to an error type that doesn't have context and then to an error message

I see. I think this would make sense if we were decoding Dna hashes in multiple places in the server, since for now we are doing it in one place, which I am not sure we will be doing, hence why I moved the error handling to the error module.